packages/kubernetes-1.24/kubelet-config

```diff
@@ -118,6 +118,10 @@ serializeImagePulls: false
 serverTLSBootstrap: {{settings.kubernetes.server-tls-bootstrap}}
 tlsCipherSuites:
 - TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
+{{#if (and settings.kubernetes.tls-certificate settings.kubernetes.tls-private-certificate)}}
+tlsCertFile: "/etc/kubernetes/pki/tls-cert.pub"
+tlsPrivateKeyFile: "/etc/kubernetes/pki/tls-cert"
+{{/if}}
```

Comment:
Does it make sense to evaluate these settings individually rather than together at once? 

`kubelet` doesn't seem to care if one or the other is missing but here we're enforcing both settings to be set for anything to be written to the kubelet config. Maybe we just render the configuration if the setting exist even if won't do anything without the other being specified?

Reply:
It might not error, but it will not use the provided files unless BOTH are provided. So it doesn't do us any good to add one without the other. You can't use the public key without the private one, and the private one will not be used to generate a public key, so it doesn't provide any benefit on its own.

There's not a lot of descriptive documentation for this, but the comments in kubelet do say this (emphasis mine):

> If tlsCertFile **AND** tlsPrivateKeyFile are not provided, a self-signed certificate and key are generated